src/api/app/controllers/webui/patchinfo_controller.rb

```diff
@@ -193,14 +193,43 @@ def save
   end
 
   def remove
+    # checks
+    if @package.name == '_project'
+      raise DeletePackageNoPermission.new '_project package can not be deleted.'
+    end
+
+    unless User.current.can_modify_package?(@package)
```

Comment:
I'd prefer to use
```
authorize package :new_incident?
```

(probably you need to introduce a policy for package ...)